archeryutils/classifications/tests/test_agb_outdoor.py

```diff
@@ -565,3 +566,187 @@ def test_calculate_agb_outdoor_classification_invalid_scores(
+                0.0,
+            ),
+            (
+                "wa720_60",
+                1,
+                "Under 18",
+                "recurve",
+                0.0,
+            ),
+        ],
+    )
+    def test_agb_outdoor_classification_fraction_low(  # noqa: PLR0913 many args
+        self,
+        score: float,
+        roundname: str,
+        age_group: str,
+        bowstyle: str,
+        frac_expected: float,
+    ) -> None:
+        """Check that classification fraction below lowest classification is 0,0."""
+        frac_returned = class_funcs.agb_outdoor_classification_fraction(
+            roundname=roundname,
+            score=score,
+            bowstyle=bowstyle,
+            gender="male",
+            age_group=age_group,
+        )
+
+        assert frac_returned == frac_expected
+
+    @pytest.mark.parametrize(
+        "roundname,score,age_group,bowstyle,frac_expected",
+        [
+            (
+                "wa720_70",
+                720,
+                "adult",
+                "compound",
+                1.0,
+            ),
+            (
+                "wa720_50_b",
+                650,
+                "adult",
+                "barebow",
+                1.0,
+            ),
+            (
+                "wa720_50_c",
+                718,
+                "adult",
+                "compound",
+                1.0,
+            ),
+            (
+                "wa720_60",
+                650,
+                "under 18",
+                "recurve",
+                1.0,
+            ),
+        ],
+    )
+    def test_agb_outdoor_classification_fraction_high(  # noqa: PLR0913 many args
+        self,
+        score: float,
+        roundname: str,
+        age_group: str,
+        bowstyle: str,
+        frac_expected: float,
+    ) -> None:
+        """Check that classification fraction above highest classification is 1,0."""
+        frac_returned = class_funcs.agb_outdoor_classification_fraction(
+            roundname=roundname,
+            score=score,
+            bowstyle=bowstyle,
+            gender="male",
+            age_group=age_group,
+        )
+
+        assert frac_returned == frac_expected
+
+    def test_agb_outdoor_classification_fraction_restrict(
+        self,
+    ) -> None:
+        """Check that classification fraction functions with restrict."""
+        frac_restricted = class_funcs.agb_outdoor_classification_fraction(
+            score=620,
+            roundname="national",
+            bowstyle="recurve",
+            gender="male",
+            age_group="adult",
+        )
+        assert frac_restricted == 1.0
+
+        frac_unrestricted = class_funcs.agb_outdoor_classification_fraction(
+            score=620,
+            roundname="national",
+            bowstyle="recurve",
+            gender="male",
+            age_group="adult",
+            restrict=False,
+        )
+
+        assert frac_unrestricted == 0.4541258975704667
```

Comment:
I can't comment on the correctness of the PR so far, but in terms of code quality it looks like you've ticked all the boxes.

The only thing bothering me is that there are a few equality tests with floats. These should probably use `assert np.isclose(x, y)`, or `np.testing.allclose(x, y)` which would give more information if the test fails.

Reply:
Sure, can do. From a quick look, I don't think it's a common issue in the tests. I see one instance where `pytest.approx` has been used, which isn't something I was aware of but it looks like it does pretty much the same thing. Maybe that would be more appropriate here?

Reply:
Good point - done with a cheeky amend and force to https://github.com/jatkinson1000/archeryutils/pull/91/commits/29fce0101540bff673e56828cc22adabdf24dd71